src/phoenix/server/api/subscriptions.py

```diff
@@ -104,18 +169,88 @@ async def chat_completion(
                     trace_rowid=trace_rowid,
                     span_id=span_id,
                     parent_id=None,
-                    name="AsyncOpenAI.chat.completion.create",
-                    span_kind="LLM",
+                    name=span_name,
+                    span_kind=LLM,
                     start_time=start_time,
                     end_time=end_time,
-                    attributes={},
-                    events=[],
-                    status_code="OK",
-                    status_message="",
-                    cumulative_error_count=0,
+                    attributes=unflatten(attributes.items()),
+                    events=finished_span.events,
+                    status_code=status.status_code.name,
+                    status_message=status.description or "",
+                    cumulative_error_count=int(not status.is_ok),
                     cumulative_llm_token_count_prompt=0,
                     cumulative_llm_token_count_completion=0,
                     llm_token_count_prompt=0,
                     llm_token_count_completion=0,
                 )
             )
+
+
+def _llm_span_kind() -> Iterator[Tuple[str, Any]]:
+    yield OPENINFERENCE_SPAN_KIND, LLM
+
+
+def _input_value_and_mime_type(input: ChatCompletionInput) -> Iterator[Tuple[str, Any]]:
+    yield INPUT_MIME_TYPE, JSON
+    yield INPUT_VALUE, json.dumps(asdict(input), cls=GraphQLInputJSONEncoder)
+
+
+def _output_value_and_mime_type(output: Any) -> Iterator[Tuple[str, Any]]:
+    yield OUTPUT_MIME_TYPE, JSON
+    yield OUTPUT_VALUE, json.dumps(output, cls=ChatCompletionOutputJSONEncoder)
+
+
+def _llm_input_messages(messages: List[ChatCompletionMessageInput]) -> Iterator[Tuple[str, Any]]:
+    for i, message in enumerate(messages):
+        yield f"{LLM_INPUT_MESSAGES}.{i}.{MESSAGE_ROLE}", message.role.value.lower()
+        yield f"{LLM_INPUT_MESSAGES}.{i}.{MESSAGE_CONTENT}", message.content
+
+
+def _llm_output_messages(content: str, role: str) -> Iterator[Tuple[str, Any]]:
+    yield f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_ROLE}", role
+    yield f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_CONTENT}", content
+
+
+def _hex(number: int) -> str:
+    """
+    Converts an integer to a hexadecimal string.
+    """
+    return hex(number)[2:]
```

Comment:
fyi, this has a difference in terms of string length.

<img width="290" alt="Screenshot 2024-10-14 at 8 27 19 AM" src="https://github.com/user-attachments/assets/ef70ddc7-6cc6-4cd9-8fca-71dc278c94e5">

Reply:
Using `number.to_bytes(8, "big")` results in `OverflowError: int too big to convert`